src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1772,6 +1772,9 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
                 var)));
         }
         std::string sym_name = x.m_name;
+        if (sym_name == "main") {
+            sym_name = "_xx_lpython_defined_main_xx";
+        }
```

Comment:
I think we should not do it here. In ASR I think we should allow any names.

Rather, in the LLVM and C backends we should rename.

Reply:
Yeah, that might be good.